In [4]:
import numpy as np
from matplotlib import pyplot as plt
import seaborn as sns
from pathlib import Path
from data_scripts import *
import pandas as pd


NUM_SIMS = 3
CONVERGED = False
USE_SIMULATED_TIME = False

In [86]:
data = {
    "Standard exploration": load_data("../1/poro_ldynamic", n_newest=NUM_SIMS),
    "No exploration": load_data("poro_eps0", n_newest=NUM_SIMS),
    "More exploration": load_data("poro_eps_big", n_newest=NUM_SIMS),
    "GP": load_data("poro_gp", n_newest=NUM_SIMS),
}

times_dict = dict()
for name, values in data.items():
    times_dict[name] = []
    for perf in values:
        res = np.sum(
            sum_per_time_step(
                make_solve_linear_system_time(perf, converged=CONVERGED),
                perf,
                converged=CONVERGED,
            )
        )
        times_dict[name].append(res)


times = pd.DataFrame.from_dict(times_dict).T
new_data = {
    'Worst': times.max(axis=1),
    'Mean': times.mean(axis=1),
    'Best': times.min(axis=1),
}
df_mandel = pd.DataFrame.from_dict(new_data)
df_mandel

Loading data:
/home/firedrake/solver_selector/examples/3.1/../1/performance/poro_ldynamic_17.npy
/home/firedrake/solver_selector/examples/3.1/../1/performance/poro_ldynamic_18.npy
/home/firedrake/solver_selector/examples/3.1/../1/performance/poro_ldynamic_19.npy
Loading data:
/home/firedrake/solver_selector/examples/3.1/performance/poro_eps0_0.npy
/home/firedrake/solver_selector/examples/3.1/performance/poro_eps0_1.npy
/home/firedrake/solver_selector/examples/3.1/performance/poro_eps0_2.npy
Loading data:
/home/firedrake/solver_selector/examples/3.1/performance/poro_eps_big_0.npy
/home/firedrake/solver_selector/examples/3.1/performance/poro_eps_big_1.npy
/home/firedrake/solver_selector/examples/3.1/performance/poro_eps_big_2.npy
Loading data:
/home/firedrake/solver_selector/examples/3.1/performance/poro_gp_0.npy
/home/firedrake/solver_selector/examples/3.1/performance/poro_gp_1.npy
/home/firedrake/solver_selector/examples/3.1/performance/poro_gp_2.npy


Worst       Mean       Best
Standard exploration  30.301274  30.014593  29.825868
No exploration        29.033704  28.609811  28.382986
More exploration      29.203206  28.974217  28.804577
GP                    28.722217  28.504058  28.351149

In [87]:
data = {
    "Standard exploration": load_data("../1/thermal_dynamic", n_newest=NUM_SIMS),
    "No exploration": load_data("thermal_eps0", n_newest=NUM_SIMS),
    "More exploration": load_data("thermal_eps_big", n_newest=NUM_SIMS),
    "GP": load_data("thermal_gp", n_newest=NUM_SIMS),
}

times_dict = dict()
for name, values in data.items():
    times_dict[name] = []
    for perf in values:
        res = np.sum(
            sum_per_time_step(
                make_solve_linear_system_time(perf, converged=CONVERGED),
                perf,
                converged=CONVERGED,
            )
        )
        times_dict[name].append(res)


times = pd.DataFrame.from_dict(times_dict).T
new_data = {
    'Worst': times.max(axis=1),
    'Mean': times.mean(axis=1),
    'Best': times.min(axis=1),
}
df_thermal = pd.DataFrame.from_dict(new_data)
df_thermal

Loading data:
/home/firedrake/solver_selector/examples/3.1/../1/performance/thermal_dynamic_31.npy
/home/firedrake/solver_selector/examples/3.1/../1/performance/thermal_dynamic_32.npy
/home/firedrake/solver_selector/examples/3.1/../1/performance/thermal_dynamic_33.npy
Loading data:
/home/firedrake/solver_selector/examples/3.1/performance/thermal_eps0_0.npy
/home/firedrake/solver_selector/examples/3.1/performance/thermal_eps0_1.npy
/home/firedrake/solver_selector/examples/3.1/performance/thermal_eps0_2.npy
Loading data:
/home/firedrake/solver_selector/examples/3.1/performance/thermal_eps_big_0.npy
/home/firedrake/solver_selector/examples/3.1/performance/thermal_eps_big_1.npy
/home/firedrake/solver_selector/examples/3.1/performance/thermal_eps_big_2.npy
Loading data:
/home/firedrake/solver_selector/examples/3.1/performance/thermal_gp_0.npy
/home/firedrake/solver_selector/examples/3.1/performance/thermal_gp_1.npy
/home/firedrake/solver_selector/examples/3.1/performance/thermal_gp_2.npy


Worst       Mean       Best
Standard exploration  77.059882  75.842509  73.667504
No exploration        75.485397  73.962586  73.041963
More exploration      73.614389  72.739651  71.663467
GP                    69.010449  67.863626  66.456043

In [88]:
df = pd.concat((df_mandel, df_thermal), join='inner', axis=1, keys=("Mandel problem", "Thermal problem"))
df = df.style.highlight_min(axis=0, props="textbf:--rwrap;").format(precision=2)

In [90]:
df

In [91]:
print(df.to_latex())

\begin{tabular}{lrrrrrr}
 & \multicolumn{3}{r}{Mandel problem} & \multicolumn{3}{r}{Thermal problem} \\
 & Worst & Mean & Best & Worst & Mean & Best \\
Standard exploration & 30.30 & 30.01 & 29.83 & 77.06 & 75.84 & 73.67 \\
No exploration & 29.03 & 28.61 & 28.38 & 75.49 & 73.96 & 73.04 \\
More exploration & 29.20 & 28.97 & 28.80 & 73.61 & 72.74 & 71.66 \\
GP & \textbf{28.72} & \textbf{28.50} & \textbf{28.35} & \textbf{69.01} & \textbf{67.86} & \textbf{66.46} \\
\end{tabular}



    \begin{tabular}{ l | rrr | rrr }
    \toprule
     & \multicolumn{3}{c}{Mandel problem} & \multicolumn{3}{c}{Thermal problem} \\
     & Worst & Mean & Best & Worst & Mean & Best \\
    Standard exploration & 30.30 & 30.01 & 29.83 & 77.06 & 75.84 & 73.67 \\
    No exploration & 29.03 & 28.61 & 28.38 & 75.49 & 73.96 & 73.04 \\
    More exploration & 29.20 & 28.97 & 28.80 & 73.61 & 72.74 & 71.66 \\
    GP & \textbf{28.72} & \textbf{28.50} & \textbf{28.35} & \textbf{69.01} & \textbf{67.86} & \textbf{66.46} \\
    \bottomrule
    \end{tabular}